kong/plugins/prometheus/exporter.lua

```diff
@@ -46,9 +46,13 @@ local function log(message)
     return
   end
 
-  local service_name = message.service and message.service.name or
-                       message.service.host
-  service_name = service_name or ""
+  local service_name
+  if message and message.service then
+    service_name = message.service.name or message.service.host
+  else
+    -- do not record any stats if the service is not present
+    return
+  end
```

Comment:
in the original code, if `message.service == {}` (no name and no host), you would get a `""`.
in this code, you will `return` if `message.service == nil`, but you will not return and get a `service_name = nil` if `message.service == {}`.

I don't know if that's a possible thing that can happen, just noting that the original caught that case and this doesn't.

If that _is_ a thing that can happen, I'd do

```lua
  local service_name = message.service and message.service.name or
                       message.service.host
  if not service_name then
     -- do not record any stats if the service is not present
     return
  end
```

Reply:
The service table comes from ngx.ctx and since host is a required field on service entity, this should not happen.
It is unlikely to happen but one rogue plugin could change ngx.ctx.service.host field to be nil.
If that does happen, it's better to probably hit a nil pointer and log an error.

Also, while typing this I recalled, having `service_name = ""` actually leads to more problems, as you will then have data in Prometheus with no meta-data.

I'm going to let this be as is for now. Thank you for catching this though!